shared/actions/chat2/index.js

```diff
@@ -809,37 +804,8 @@ const loadMoreMessages = (
         return arr
       }, [])
 
-      // Still loading this conversation w/ this paginationKey?
-      if (
-        _loadingMessagesWithPaginationKey[Types.conversationIDKeyToString(conversationIDKey)] ===
-        paginationKey
-      ) {
-        let newPaginationKey = Types.stringToPaginationKey(
-          (uiMessages.pagination && uiMessages.pagination.next) || ''
-        )
-
-        if (context === 'full') {
-          const paginationMoreToLoad = uiMessages.pagination ? !uiMessages.pagination.last : true
-          // if last is true on the full payload we blow away paginationKey
-          newPaginationKey = paginationMoreToLoad ? newPaginationKey : Types.stringToPaginationKey('')
-        }
-        yield Saga.put(
-          Chat2Gen.createMetaUpdatePagination({conversationIDKey, paginationKey: newPaginationKey})
-        )
-      }
-
-      // If we're loading the thread clean lets clear
-      if (!calledClear && action.type !== Chat2Gen.loadOlderMessagesDueToScroll) {
-        calledClear = true
-        // only clear if we've never seen the oldest message, implying there is a gap
-        if (messages.length) {
-          const oldestOrdinal = messages[messages.length - 1].ordinal
```

Comment:
I think this line in particular is a trouble spot, I should have spotted it earlier. The problem here is all you need to do to satisfy this condition is receive a new incoming message with the right message ID. That is, let's say you receive messages 1-50, but then miss every message 50-99, but then come online and receive message ID 100. Now suppose you click into this thread and load up messages 100-120. The app will now scroll up into messages 1-50.